src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -169,6 +174,12 @@ public DynamicObject mulOther(DynamicObject array, int count,
             return result;
         }
 
+        @TruffleBoundary
```

Comment:
No boundary when throwing RaiseException, Truffle needs to see it to optimize it.